include/fiction/algorithms/simulation/sidb/defect_operational_domain.hpp

```diff
@@ -0,0 +1,812 @@
+
+    if (stats)
+    {
+        *stats = st;
+    }
+
+    return result;
+}
+/**
+ * Computes the defect operational domain of the given SiDB cell-level layout. The defect operational domain is the
+ * set of all defect positions for which the layout is logically operational. Logical operation is defined as the layout
+ * implementing the given truth table. The input BDL pairs of the layout are assumed to be in the same order as the
+ * inputs of the truth table.
+ *
+ * This algorithm uses random sampling to find a part of the defect operational domain that might not be
+ * complete. It performs a total of samples uniformly-distributed random samples within the specified area.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @tparam TT Truth table type.
+ * @param lyt Layout to compute the defect operational domain for.
+ * @param spec Expected Boolean function of the layout given as a multi-output truth table.
+ * @param samples Number of random samples to perform.
+ * @param params Defect operational domain computation parameters.
+ * @param stats Statistics.
+ * @return The (partial) defect operational domain of the layout.
+ */
+template <typename Lyt, typename TT>
+defect_operational_domain<Lyt>
+defect_operational_domain_random_sampling(const Lyt& lyt, const std::vector<TT>& spec, std::size_t samples,
+                                          const defect_operational_domain_params& params = {},
+                                          defect_operational_domain_stats*        stats  = nullptr)
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+    static_assert(kitty::is_truth_table<TT>::value, "TT is not a truth table");
+    static_assert(has_cube_coord_v<Lyt>, "Lyt is not based on cube coordinates");
+
+    defect_operational_domain_stats                 st{};
+    detail::defect_operational_domain_impl<Lyt, TT> p{lyt, spec, params, st};
+
+    const auto result = p.random_sampling(samples);
+
+    if (stats)
+    {
+        *stats = st;
+    }
+
+    return result;
+}
+
+/**
+ * Computes the defect operational domain of the given SiDB cell-level layout. The defect
+ * operational domain is the set of all defect positions for which the layout is logically operational. Logical
+ * operation is defined as the layout implementing the given truth table. The input BDL pairs of the layout are assumed
+ * to be in the same order as the inputs of the truth table.
+ *
+ * This algorithm uses contour tracing to identify operational defect locations within the SiDB layout.
+ * It starts by searching for defect locations on the left side (bounding_box + additional scanning area). The
+ * y-coordinate for these positions is chosen randomly. The number of samples is determined by the `samples` parameter.
+ *
+ * Then the algorithm moves each defect position to the right, searching for the last operational defect position. This
```

Comment:
```suggestion
 * Then, the algorithm moves each defect position to the right, searching for the last operational defect position. This
```
Also, what does "last" mean here?